libraries/chain/steem_evaluator.cpp

```diff
@@ -2025,130 +2025,129 @@ namespace golos { namespace chain {
-                          o.recovery_account, "Cannot recover an account that does not have you as there recovery partner.");
-            else                                                  // Empty string recovery account defaults to top witness
-                FC_ASSERT(
-                        _db.get_index<witness_index>().indices().get<by_vote_name>().begin()->owner ==
-                        o.recovery_account, "Top witness must recover an account with no recovery partner.");
+        void request_account_recovery_evaluator::do_apply(const request_account_recovery_operation& o) {
+            const auto& account_to_recover = _db.get_account(o.account_to_recover);
+            if (account_to_recover.recovery_account.length()) {
+                // Make sure recovery matches expected recovery account
+                GOLOS_CHECK_LOGIC(account_to_recover.recovery_account == o.recovery_account,
+                    logic_exception::cannot_recover_if_not_partner,
+                    "Cannot recover an account that does not have you as there recovery partner.");
+            } else {
+                // Empty string recovery account defaults to top witness
+                GOLOS_CHECK_LOGIC(
+                    _db.get_index<witness_index>().indices().get<by_vote_name>().begin()->owner == o.recovery_account,
+                    logic_exception::must_be_recovered_by_top_witness,
+                    "Top witness must recover an account with no recovery partner.");
+            }
 
-            const auto &recovery_request_idx = _db.get_index<account_recovery_request_index>().indices().get<by_account>();
+            const auto& recovery_request_idx =
+                _db.get_index<account_recovery_request_index>().indices().get<by_account>();
             auto request = recovery_request_idx.find(o.account_to_recover);
 
-            if (request == recovery_request_idx.end()) // New Request
-            {
-                FC_ASSERT(!o.new_owner_authority.is_impossible(), "Cannot recover using an impossible authority.");
-                FC_ASSERT(o.new_owner_authority.weight_threshold, "Cannot recover using an open authority.");
-
+            if (request == recovery_request_idx.end()) {
+                // New Request
+                GOLOS_CHECK_OP_PARAM(o, new_owner_authority, {
+                    GOLOS_CHECK_VALUE(!o.new_owner_authority.is_impossible(),
+                        "Cannot recover using an impossible authority.");
+                    GOLOS_CHECK_VALUE(o.new_owner_authority.weight_threshold,
+                        "Cannot recover using an open authority.");
+                });
                 // Check accounts in the new authority exist
-                if ((_db.has_hardfork(STEEMIT_HARDFORK_0_15__465) ||
-                     _db.is_producing())) {
-                    for (auto &a : o.new_owner_authority.account_auths) {
+                if (_db.has_hardfork(STEEMIT_HARDFORK_0_15__465) || _db.is_producing()) {
```

Comment:
`_db.is_producing()` can be removed from here

Reply:
Yeah, you're right, fixed this and line 2076